File: src/consumer.js
const Pulsar = require('pulsar-client');

(async () => {
  // Token based on authentication
  const tokenStr = '{JWT token}';
	const pulsarUri = 'pulsar+ssl://{host}:6651';
	const topicName = 'persistent://{tenant}/{namespace}/{topic}';

	const trustStore = '/etc/ssl/certs/ca-bundle.crt';
  // Debian Ubuntu:
  // const trustStore = '/etc/ssl/certs/ca-certificates.crt'
  // OSX:
  // Export the default certificates to a file, then use that file:
  // security find-certificate -a -p /System/Library/Keychains/SystemCACertificates.keychain > ./ca-certificates.crt
  // trust_certs='./ca-certificates.crt'

  const auth = new Pulsar.AuthenticationToken({token: tokenStr});

  // Create a client
  const client = new Pulsar.Client({
    serviceUrl: pulsarUri,
    authentication: auth,
    tlsTrustCertsFilePath: trustStore,
    operationTimeoutSeconds: 30,
  });

  // Create consumer
  const consumer = await client.subscribe({
    topic: topicName,
    subscription: 'my-subscription',
    subscriptionType: 'Exclusive',
    ackTimeoutMs: 10000,
  });

  // Receive and acknowledge messages
  for (let i = 0; i < 1000; i += 1) {
    const msg = await consumer.receive();
    console.log(msg.getData().toString());
    consumer.acknowledge(msg);
  }

  await consumer.close();
  await client.close();
})();

File: cpp/consumer.cc
#include <iostream>

#include <pulsar/Client.h>

using namespace pulsar;
using namespace std;

int main() {

    std::string token = "eyJhbGciOiJ Pulsar JWT";
    std::string serviceURL = "pulsar+ssl://useast2.aws.kafkaesque.io:6651";
    std::string topicName = "persistent://ming-luo/local-useast2-aws/test-topic2";

    AuthenticationPtr auth = pulsar::AuthToken::createWithToken(token);
    ClientConfiguration config = ClientConfiguration();
    config.setAuth(auth);
    config.setTlsTrustCertsFilePath("/etc/ssl/certs/ca-bundle.crt");
    /**
     *Use default CA certs for your environment
     * RHEL/CentOS:
     * trust_certs='/etc/ssl/certs/ca-bundle.crt'
     * Debian/Ubuntu:
     * trust_certs='/etc/ssl/certs/ca-certificates.crt'
     */
    Client client(serviceURL, config);

    Consumer consumer;
    Result result = client.subscribe(topicName, "consumer-name", consumer);
    if (result != ResultOk) {
        cout << "Failed to subscribe: " << result << endl;
        return -1;
    }

    Message msg;

    while (true) {
        consumer.receive(msg);
        cout << "Received: " << msg << "  with payload '" << msg.getDataAsString() << "'" << endl;

        consumer.acknowledge(msg);
    }

    client.close();
}

